meta.go

```diff
@@ -92,12 +93,30 @@ type Storage struct {
 	// CountMax defaults to 1 for singleton stores.
 	CountMax int `bson:"countmax"`
 
+	// MinimumSize is the minimum size of store that the charm needs to
+	// work at all. This is not a recommended size or a comfortable size
+	// or a will-work-well size, just a bare minimum below which the charm
+	// is going to break.
+	// MinimumSize requires a unit, one of MGTPEZY, and is stored as MiB.
+	//
+	// There is no default MinimumSize; if left unspecified, a provider
+	// specific default will be used, typically 1GB for block storage.
+	MinimumSize uint64 `bson:"minimum-size"`
+
 	// Location is the mount location for filesystem stores. For multi-
 	// stores, the location acts as the parent directory for each mounted
 	// store.
 	//
 	// Location has no default, and is optional.
 	Location string `bson:"location,omitempty"`
+
+	// Properties allow the charm author to characterise the relative storage
+	// performance requirements and sensitivities for each store.
+	// eg “transient” is used to indicate that non persistent storage is acceptable,
+	// such as tmpfs or ephemeral instance disks.
+	//
+	// Properties has no default, and is optional.
+	Properties []string `bson:properties,omitempty`
```

Comment:
As discussed live, the properties are used as hints and jam is ok with implementing these now.